xonsh/commands_cache.py

```diff
@@ -73,7 +73,14 @@ def get_possible_names(name):
     @property
     def all_commands(self):
         paths = builtins.__xonsh_env__.get('PATH', [])
-        pathset = frozenset(x for x in paths if os.path.isdir(x))
+        def removeDups(p):
```

Comment:
We (and PEP8) prefer snake_case for function names, can this be `remove_dups` please?

Reply:
Also, can you please move this function to be a method on the class or to part of the `xonsh.tools` module?  `all_commands` is a performance ciritical pathway and so we don't want to redefine this function each time.